vm/src/vm/vm_core.rs

```diff
@@ -905,6 +905,10 @@ impl VirtualMachine {
         self.trace = None
     }
 
+    pub fn get_relocation_table(&self) -> Result<Vec<usize>, MemoryError> {
```

Comment:
Why are we adding this?
I think it is not necessary, plus the get_ method name can be misunderstood since it is not just a get, it relocates the memory

Reply:
So the trader needed the relocated segments but `segments` was public only inside the crate. I have renamed the function now to `relocate_segments` and added a `#[cfg(feature = "with_tracer")]` flag as well.